File: LeetCode 75/1207. Unique Number of Occurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        HashMap<Integer, Integer > map = new HashMap<>();
        for (int i : arr) {
            map.put(i, map.getOrDefault(i,0)+1);
        }
        HashSet<Integer> hset=new HashSet<>(map.values());

        return hset.size() == map.size();
    }
}

File: LeetCode 75/1207. Unique Number of Occurrences.kt
fun uniqueOccurrences(arr: IntArray) : Boolean{
    val  map = mutableMapOf<Int, Int>()

    arr.forEach {
        map.put(it, (map.get(it) ?: 0 ) + 1)
    }

    return map.values.size == map.values.toSet().size
    
}
